ui/components/app/signature-request-original/signature-request-original.component.js

```diff
@@ -258,13 +241,60 @@ export default class SignatureRequestOriginal extends Component {
     );
   };
 
-  renderFooter = () => {
+  renderWarning = () => {
     const {
       cancel,
       clearConfirmTransaction,
       history,
       mostRecentOverviewPage,
       sign,
+    } = this.props;
+    const { fromAccount } = this.state;
+    const { trackEvent, type } = this.context;
+
+    return (
+      <SignatureRequestOriginalWarning
+        senderAddress={fromAccount.address}
+        name={fromAccount.name}
+        onSubmit={async (event) => {
+          await sign(event);
+          trackEvent({
+            category: EVENT.CATEGORIES.TRANSACTIONS,
+            event: 'Confirm',
+            properties: {
+              action: 'Sign Request',
+              legacy_event: true,
+              type,
+            },
+          });
+          clearConfirmTransaction();
+          history.push(mostRecentOverviewPage);
+        }}
+        onCancel={async (event) => {
+          await cancel(event);
+          trackEvent({
+            category: EVENT.CATEGORIES.TRANSACTIONS,
+            event: 'Cancel',
+            properties: {
+              action: 'Sign Request',
+              legacy_event: true,
+              type,
+            },
+          });
```

Comment:
@bschorchit might be a good opportunity to work with @worldlyjohn to get this inline for new schema?

Reply:
Non blocking, could be spun out into a new ticket.

Reply:
That is probably correct, actually. for some reason I was thinking these events were for the warning not the signature request itself.

Reply:
Generally prefer seeing events as close to the user action that triggered it. By that I mean keeping these and removing the implementation elsewhere, but I realize that's likely out of scope (right?) so then I'd agree with comments above to remove this.

Reply:
I've created this issue https://github.com/MetaMask/metamask-extension/issues/16617 regarding the removal of `Confirm` and `Cancel` events